sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -227,6 +227,11 @@ class Dataset[T] private[sql](
       dsIds.add(id)
       plan.setTagValue(Dataset.DATASET_ID_TAG, dsIds)
     }
+    // A plan might get its PLAN_ID_TAG via connect or belong to multiple dataframes so only assign
+    // an id to a plan if it doesn't have any
+    if (plan.getTagValue(LogicalPlan.PLAN_ID_TAG).isEmpty) {
```

Comment:
IIUC, this PR aims to fix a vanilla (non-Connect) Spark SQL issue with `PLAN_ID_TAG`.
However, currently the `PLAN_ID_TAG` is only dedicated for Spark Connect.

Reply:
Yes, that idea came up here: https://github.com/apache/spark/pull/45446#discussion_r1520922028